### PR TITLE
chore(core): update documentation in transform logic and processor 🙀

### PR DESCRIPTION
**Pull request by srl295**
- drop some unneeded variables also, should be no semantic change

For: #7375

@keymanapp-test-bot skip

### CODE REVIEW

File: core/src/ldml/ldml_processor.cpp [hunk 1]
```diff
@@ -258,16 +259,16 @@ ldml_processor::process_event(
         // Look up the key
         const std::u16string str = keys.lookup(vk, modifier_state);
         if (str.empty()) {
-          // not found
+          // not found, so pass the keystroke on to the
```
Comment (mcdurdin) [inline core/src/ldml/ldml_processor.cpp:262]:
```suggestion
          // not found, so pass the keystroke on to the Engine
```

File: core/src/ldml/ldml_processor.cpp [hunk 2]
```diff
@@ -258,16 +259,16 @@ ldml_processor::process_event(
...
           break; // ----- commit and exit
         }
         // found the correct string - push it into the context and actions
         const std::u32string str32 = kmx::u16string_to_u32string(str);
-        for(size_t i=0; i<str32.length(); i++) {
-          state->context().push_character(str32[i]);
-          state->actions().push_character(str32[i]);
+        for (const auto &ch : str32) {
+          state->context().push_character(ch);
+          state->actions().push_character(ch);
```
Comment (mcdurdin) [inline core/src/ldml/ldml_processor.cpp:271]: The potential for desync between actions and context is a real design problem ... not to fix here but one day we should discuss

**Review by mcdurdin — APPROVED** [review]
LGTM

File: core/src/ldml/ldml_processor.cpp (hunk 2 above)
Comment (srl295) [reply core/src/ldml/ldml_processor.cpp:271]: I'll need to learn how that works.. but sgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.149-alpha](https://keyman.com/downloads/releases/alpha/17.0.149)

**Pull request merged**